### PR TITLE
Feat: allow setting editorial workflow PR label prefix

### LINKED ISSUE #3977: Add MR/PR Label Prefix Config Option
**Is your feature request related to a problem? Please describe.**
I'm using GitLab with scoped labels for my projects. As the suggestion is to name labels like `scope::label` I would like to be able to use the same style for my netlify-cms editorial workflow labels. At the moment the prefix for the editorial workflow labels is hard-coded to `netlify-cms/`.

**Describe the solution you'd like**
I would suggest a configuration option at least for the hosted Git Backends (GitLab, GitHub, Bitbucket) named `label_prefix` that is `netlify-cms/` per default but can be adjusted.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by KancerEzeroglu**
closes #3977 

**Summary**
When you are using editorial workflow, status labels are created with hardcoded `netlify-cms/` prefix. 
With this PR, we enable you to add a custom prefix to config.yml which overwrites the default hardcoded one.
I only did changes for GitHub. If I am in the right direction, I'll make changes for others (GitLab, bitbucket) as well.

**Test plan**
Unit tests

**A picture of a cute animal (not mandatory but encouraged)**
my first PR 👯 

firsttime

### CODE REVIEW

File: packages/netlify-cms-lib-util/src/APIUtils.ts [hunk 1]
```diff
@@ -2,10 +2,15 @@ export const CMS_BRANCH_PREFIX = 'cms';
...
 export const MERGE_COMMIT_MESSAGE = 'Automatically generated. Merged on Netlify CMS.';
 
-const NETLIFY_CMS_LABEL_PREFIX = 'netlify-cms/';
-export const isCMSLabel = (label: string) => label.startsWith(NETLIFY_CMS_LABEL_PREFIX);
-export const labelToStatus = (label: string) => label.substr(NETLIFY_CMS_LABEL_PREFIX.length);
-export const statusToLabel = (status: string) => `${NETLIFY_CMS_LABEL_PREFIX}${status}`;
+const DEFAULT_NETLIFY_CMS_LABEL_PREFIX = 'netlify-cms/';
+const getLabelPrefix = (labelPrefix: string) => labelPrefix || DEFAULT_NETLIFY_CMS_LABEL_PREFIX;
+
+export const isCMSLabel = (label: string, labelPrefix = '') =>
```
Comment (erezrokah) [inline packages/netlify-cms-lib-util/src/APIUtils.ts:8]: I recommend removing the default value for `labelPrefix`, then running `yarn test` to see all the places you should be passing the prefix

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @KancerEzeroglu. Nice work.
There are still some places you need to pass the prefix (mostly for non GitHub backends).

File: dev-test/config.yml [hunk 2]
```diff
@@ -245,3 +245,14 @@ collections: # A list of collections the CMS should be able to edit
               - { label: 'Date', name: 'date', widget: 'date' }
               - { label: 'Image', name: 'image', widget: 'image' }
               - { label: 'File', name: 'file', widget: 'file' }
+  - name: pages # a nested collection
```
[the remark this answers is not in the document]
Comment (KancerEzeroglu) [reply dev-test/config.yml:248]: I removed this part.

**Comment by erezrokah** [comment]
@KancerEzeroglu, I moved the config under the `backend` section, added docs and added more explicit names to some tests.

**Comment by KancerEzeroglu** [comment]
Thanks @erezrokah 😄 
I will review the changes you have made to better understand the code and see where I am missing.

**Pull request merged**